metatensor-core/include/metatensor.h

```diff
@@ -831,6 +844,87 @@ struct mts_tensormap_t *mts_tensormap_keys_to_samples(const struct mts_tensormap
                                                       struct mts_labels_t keys_to_move,
                                                       bool sort_samples);
 
+/**
+ * Load labels from the file at the given path.
+ *
+ * Labels are stored using numpy's NPY format, so the file will typically use
+ * the `.npy` extension.
+ *
+ * This function allocates memory which must be released `mts_labels_free` when
+ * you don't need it anymore.
+ *
+ * @param path path to the file as a NULL-terminated UTF-8 string
+ * @param labels pointer to empty Labels
+ * @returns The status code of this operation. If the status is not
+ *          `MTS_SUCCESS`, you can use `mts_last_error()` to get the full
+ *          error message.
+ */
+mts_status_t mts_labels_load(const char *path, struct mts_labels_t *labels);
+
+/**
+ * Load labels from the given in-memory buffer.
+ *
+ * This function allocates memory which must be released `mts_labels_free` when
+ * you don't need it anymore.
+ *
+ * @param buffer buffer containing a previously serialized `mts_labels_t`
+ * @param buffer_count number of elements in the buffer
+ * @param labels pointer to empty Labels
+ * @returns The status code of this operation. If the status is not
+ *          `MTS_SUCCESS`, you can use `mts_last_error()` to get the full
+ *          error message.
+ */
+mts_status_t mts_labels_load_buffer(const uint8_t *buffer,
+                                    uintptr_t buffer_count,
+                                    struct mts_labels_t *labels);
+
+/**
+ * Save labels to the file at the given path.
+ *
+ * If the file already exists, it is overwritten.
+ *
+ * @param path path to the file as a NULL-terminated UTF-8 string
+ * @param labels Labels to save to the file
+ *
+ * @returns The status code of this operation. If the status is not
+ *          `MTS_SUCCESS`, you can use `mts_last_error()` to get the full
+ *          error message.
+ */
+mts_status_t mts_labels_save(const char *path, struct mts_labels_t labels);
+
+/**
+ * Save labels to an in-memory buffer.
+ *
+ * On input, `*buffer` should contain the address of a starting buffer (which
+ * can be NULL) and `*buffer_count` should contain the size of the allocation
+ * (0 if `*buffer` is NULL).
+ *
+ * On output, `*buffer` will contain the serialized data, and `*buffer_count`
+ * the total number of written bytes (which might be less than the allocation
+ * size).
```

Comment:
So it allocates its own memory if it's passed NULL?

Reply:
If `buffer` starts as NULL, `realloc` should allocate memory for it yes!